examples/schema.rs

```diff
@@ -0,0 +1,137 @@
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    if schema.is_ok() {
+        eprintln!("Schema: {:?} was successfully loaded", schema.unwrap().id());
+    } else {
+        eprintln!("{:?}", schema.unwrap_err());
+    }
+
+    Ok(())
+}
+
+fn validate(command_args: &ArgMatches) -> IonSchemaResult<()> {
+    // Extract the user provided authorities
+    let authorities: Vec<_> = command_args.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = command_args.value_of("schema").unwrap();
+
+    // Extract Ion value provided by user
+    let input_file = command_args.value_of("input").unwrap();
+    let value = fs::read(input_file).expect("Can not load given ion file");
+    let owned_elements = element_reader()
+        .read_all(&value)
+        .expect("parsing failed unexpectedly");
+
+    // Set up authorities vector
+    let mut document_authorities: Vec<Box<dyn DocumentAuthority>> = vec![];
+
+    for authority in authorities {
+        document_authorities.push(Box::new(FileSystemDocumentAuthority::new(Path::new(
+            authority,
+        ))))
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    // get the type defined within the schema file
+    // there will only be a single type defined within schema
+    let type_ref = schema
+        .unwrap()
+        .get_types()
+        .next()
+        .expect("Loaded schema was empty");
+
+    // create a text writer to make the output
+    let mut output = vec![];
+    let mut writer = TextWriter::new(&mut output);
+
+    // validate owned_elements according to type_ref
+    for owned_element in owned_elements {
+        // create a validation report with validation result, value, schema and/or violation
+        writer.step_in(IonType::Struct)?;
+        let validation_result = type_ref.validate(&owned_element);
+        writer.set_field_name("result");
+        match validation_result {
+            Ok(_) => {
+                writer.write_string("Valid")?;
+                writer.set_field_name("value");
+                const TEST_BUF_LEN: usize = 4 * 1024 * 1024;
+                let mut buf = vec![0u8; TEST_BUF_LEN];
+                let mut element_writer =
+                    Format::Text(TextKind::Pretty).element_writer_for_slice(&mut buf)?;
+                element_writer.write(&owned_element)?;
+                let slice = element_writer.finish()?;
+                let slice = from_utf8(slice).unwrap_or("<INVALID UTF-8>");
+                writer.write_string(slice)?;
+                writer.set_field_name("schema");
+                writer.write_string(schema_id)?;
```

Comment:
Consider adding a `type` field even though we expect there to only be one type in the schema. (The current implementation uses the first type it finds, but doesn't raise an error if there's more than one.)

Reply:
Added a new `type` field for specifying particular type for validation.